misc/snapshotter/Dockerfile

```diff
@@ -1,12 +1,21 @@
 FROM alpine:3.17.0 AS sourcer
 
+ARG TARGETARCH
+
 RUN apk add --no-cache curl
 RUN apk add --no-cache --upgrade grep
-RUN export NYDUS_VERSION=$(curl --silent "https://api.github.com/repos/dragonflyoss/image-service/releases/latest" | grep -Po '"tag_name": "\K.*?(?=")') && \
-    wget https://github.com/dragonflyoss/image-service/releases/download/$NYDUS_VERSION/nydus-static-$NYDUS_VERSION-linux-amd64.tgz && \
+RUN unset NYDUS_VERSION && \
+    while [ -z $NYDUS_VERSION ]; do \
+        export NYDUS_VERSION=$(curl --silent "https://api.github.com/repos/dragonflyoss/image-service/releases/latest" | grep -Po '"tag_name": "\K.*?(?=")'); sleep 1; \
+    done && \
+    echo NYDUS_VERSION: $NYDUS_VERSION && \
+    if [ -z $NYDUS_VERSION ]; then echo Invalid NYDUS_VERSION && exit -1; fi && \
+    echo TARGETARCH: $TARGETARCH && \
+    if [ "$TARGETARCH" != "amd64" -a "$TARGETARCH" != "arm64" -a "$TARGETARCH" != "ppc64le" -a "$TARGETARCH" != "riscv64" ]; then echo Unsupported TARGETARCH for nydus && exit -1; fi && \
```

Comment:
I still incline to simplify the test by assuming that only amd64 is the base nydus artifacts platform since the test target platform is decided and can't changed